broker.go

```diff
@@ -12,6 +14,9 @@ type Broker interface {
 
 // A super simple broker that matches URIs to Subscribers.
 type defaultBroker struct {
+	optionsLock   sync.RWMutex
+	routesLock    sync.RWMutex
+	subsLock      sync.RWMutex
```

Comment:
This feels like overkill, only because of the number of changes to the original source.  I don't think it would be too big of a problem to use a single `RWMutex` to protect the broker.  It would be read-locked in `Publish` and write-locked in `Subscribe` and `Unsubscribe`.  These last two are relatively rare compared to the former -- and also faster to execute.